crates/rune/src/runtime/vm.rs

```diff
@@ -984,14 +1022,33 @@ impl Vm {
             }
             target => {
                 let hash = field.hash();
-
-                match self.call_field_fn(Protocol::SET, target, hash, (value,))? {
-                    CallResult::Ok(()) => {
-                        self.stack.pop()?;
-                        CallResult::Ok(())
+                let type_hash =
+                    Hash::field_fn(Protocol::SET, target.type_hash()?, hash.into_type_hash());
+                return Ok(if let Some(handler) = self.context.function(type_hash) {
+                    match unsafe {
+                        //Unsafe block required to avoid cloning index if it isn't needed.
```

Comment:
Rework this to get rid of `unsafe` even if it requires cloning some things. Coercing an `&mut self` while borrowing the handler is not appropriate and might result in non-local unsoundness if the implementation was ever changed to do something differently. This should also be using `Vm::call_field_fn`.

Reply:
that is fine, however the reason I used an alternative implementation to call_field_fn is due to fact call_field_fn did extra work which seemed wasteful, but I am happy to change it and rerun the benchmarks.

regarding having a type having both `GET` and `INDEX_GET` does not necessarily seem like a issue overall, however if you would prefer it could be possibly to have types opt into the fallback functionality?

Reply:
I have separated the Protocols from INDEX_GET and INDEX_SET to FALLBACK_GET and FALLBACK_SET.